src/platform/src/main/java/org/geoserver/platform/GeoServerEnvironment.java

```diff
@@ -120,6 +121,33 @@ protected Properties parseFileContents(InputStream in) throws IOException {
         }
     }
 
+    private void resolvePropertyLoader(String propertyFile) {
+        GeoServerResourceLoader loader = GeoServerExtensions.bean(GeoServerResourceLoader.class);
```

Comment:
This method and the other below are very similar could you refact this to avoid code duplication?